### DISCUSSION TITLE
Why does piping of CLSID to start fail?

### DISCUSSION THREAD

**Discussion by sdbbs** [PowerShell/PowerShell#21541] [Q&A]
In writing https://stackoverflow.com/questions/78392559/how-to-create-a-windows-control-panel-window-using-clsid-in-powershell , I arrived at this snippet:

```
PS C:\tmp> "::{7007ACC7-3202-11D1-AAD2-00805FC1270E}" | start

cmdlet Start-Process at command pipeline position 1
Supply values for the following parameters:
FilePath: v
start : This command cannot be run due to the error: The system cannot find the file specified.
At line:1 char:46
+ "::{7007ACC7-3202-11D1-AAD2-00805FC1270E}" | start
+                                              ~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Start-Process], InvalidOperationException
    + FullyQualifiedErrorId : InvalidOperationException,Microsoft.PowerShell.Commands.StartProcessCommand
```

Note that the "usual", non-piped command:

```
PS C:\tmp> start "::{7007ACC7-3202-11D1-AAD2-00805FC1270E}"
```

... works fine, and starts the (Control Panel\) Network Connection windows.

Why does piping CLSID to `start` fail?

**Comment by rhubarb-geek-nz** [comment]
Hi,

[#region StartProcessCommand](https://github.com/PowerShell/PowerShell/blob/5efd627e91325a0d51df0167bd08609fa85acb56/src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs#L1598)

I don't see any ValueFromPipeline or similar attributes on any of the parameters so I assume PowerShell does not know what to do with the input pipeline. It has to populate one of the parameters for ProcessRecord() to find it.

Although it may fail to start, it did not start to fail, or did it previously work and this is a change in behaviour, or just expectations?

**Comment by ghost** [comment]
Although it may fail to start, it did not start to fail, or did it previously work and this is a change in behaviour, or just expectations?
